packages/state-transition/src/epoch/processPendingDeposits.ts

```diff
@@ -29,6 +29,13 @@ export function processPendingDeposits(state: CachedBeaconStateElectra, cache: E
   // TODO: is this a good number?
   const chunk = 100;
   const pendingDepositsLength = state.pendingDeposits.length;
+
+  if (pendingDepositsLength === 0) {
+    // Need to handle length = 0 case because state.pendingDeposits.getReadonlyByRange will throw error
```

Comment:
if `pendingDepositsLength == 0` then it'll never go to the below `while` because `startIndex` is also 0

Reply:
> if `pendingDepositsLength == 0` then it'll never go to the below `while` because `startIndex` is also 0

Yea you are right.

I encountered 
```Error: Error getting by range, startIndex >= length: 0 >= 0```
in the spec test of #7598 probably because of `dirtyLength` and this will not fix it.